build-system/app-index/template.js

```diff
@@ -24,8 +24,33 @@ const documentModes = require('./document-modes');
 const html = require('./html');
 const ProxyForm = require('./proxy-form');
 const {KeyValueOptions} = require('./form');
+const {many} = require('./html-helpers');
 const {SettingsModal, SettingsOpenButton} = require('./settings');
 
+const fileListEndpointPrefix = '/dashboard/api/listing';
+
+const examplesPathRegex = /^\/examples\//;
+const htmlDocRegex = /\.html$/;
+
+const leadingSlashRegex = /^\//;
+
+const replaceLeadingSlash = (subject, replacement) =>
+  subject.replace(leadingSlashRegex, replacement);
+
+const fileListEndpoint = query =>
+  fileListEndpointPrefix + '?' +
+  Object.keys(query).map(k => `${k}=${query[k]}`).join('&');
+
+const ampStateKey = (...keys) => keys.join('.');
+
+const selectModeStateId = 'documentMode';
+const selectModeStateKey = 'selectModePrefix';
```

Comment:
Same for these types of variables. Can they ben in a constants object? 😄

Reply:
Ditto as above.